### PR TITLE
Update pico-sdk to 1.5.1

### PR DESCRIPTION
**Pull request by bill88t**
I am primarily looking at:
`cyw43_arch_disable_sta_mode()`
`cyw43_arch_disable_ap_mode()`

Just note, I have yet to test it.
I will let you know once I have tested everything.

### CODE REVIEW

**Comment by bill88t** [comment]
https://github.com/raspberrypi/pico-sdk/pull/1435 is needed. Otherwise we would have to remove our `-Wundef`.

**Comment by jepler** [comment]
We have an existing line of code that sets different flags for SDK files (including -Wno-undef) but it evidently is not being applied to this file. It may be because it is a ".S" file or for another reason I'm not aware of:
```make
SRC_SDK := $(addprefix sdk/, $(SRC_SDK))
$(patsubst %.c,$(BUILD)/%.o,$(SRC_SDK) $(SRC_CYW43)): CFLAGS += -Wno-missing-prototypes -Wno-undef -Wno-unused-function -Wno-nested-externs -Wno-strict-prototypes -Wno-double-promotion -Wno-sign-compare -Wno-unused-variable -Wno-strict-overflow -Ilib/cyw43-driver
```

This line is acting only on ".c" files listed in SRC_SDK, it would need to act on ".S" files as well. These are listed in SRC_C_UPPER. So it might be something like
```patch
diff --git a/ports/raspberrypi/Makefile b/ports/raspberrypi/Makefile
index f23da5a51d..9a7d1b69b7 100644
--- a/ports/raspberrypi/Makefile
+++ b/ports/raspberrypi/Makefile
@@ -236,6 +236,7 @@ SRC_SDK := \
 
 SRC_SDK := $(addprefix sdk/, $(SRC_SDK))
 $(patsubst %.c,$(BUILD)/%.o,$(SRC_SDK) $(SRC_CYW43)): CFLAGS += -Wno-missing-prototypes -Wno-undef -Wno-unused-function -Wno-nested-externs -Wno-strict-prototypes -Wno-double-promotion -Wno-sign-compare -Wno-unused-variable -Wno-strict-overflow -Ilib/cyw43-driver
+$(patsubst %.S,$(BUILD)/%.o,$(SRC_C_CUPPER) $(SRC_CYW43)): CFLAGS += -Wno-undef
 
 SRC_C += \
 	boards/$(BOARD)/board.c \
```

**Comment by tannewt** [comment]
Do we want to wait until after 8.2 to do this?

**Comment by bill88t** [comment]
If that is the case, `stop_ap` for picow is gonna be only on 9.0 since no 8.3. Your choice.
I will complete the PR asap.

**Comment by bill88t** [comment]
It now builds, but I saw some instability in wifi connection code (I once got a `ConnectionError: Unknown failure 1` right after boot). It will need some testing. It works a good bit though.

File: ports/raspberrypi/Makefile [hunk 1]
```diff
@@ -171,7 +172,7 @@ else
 endif
 
 # Remove -Wno-stringop-overflow after we can test with CI's GCC 10. Mac's looks weird.
-DISABLE_WARNINGS = -Wno-stringop-overflow -Wno-cast-align
+DISABLE_WARNINGS = -Wno-stringop-overflow -Wno-cast-align -Wno-undef
```
Comment (jepler) [inline ports/raspberrypi/Makefile]: We prefer to keep our code free of `-Wundef` diagnostics by leaving the flag as-is for most files. My [comment](https://github.com/adafruit/circuitpython/pull/8091#issuecomment-1599320552) suggested how to apply the `-Wno-undef` compiler flag only to files within the SDK. Did you try this?

File: ports/raspberrypi/Makefile [hunk 2]
```diff
@@ -142,6 +142,7 @@ INC += \
         -Isdk_config \
         -I../../lib/tinyusb/src \
         -I../../supervisor/shared/usb \
+        -I../bindings/cyw43/__init__.h \
```
[the remark this answers is not in the document]
Comment (jepler) [reply ports/raspberrypi/Makefile]: [whoops, this and the other comment are outdated; I selected "cancel review" but didn't realize it'd still post these comments]

**Comment by jepler** [comment]
@dhalbert @bill88t I think this is ready for review now. It looks good to me but I'm not marking it "approved" since I made changes in the branch & I want to make sure it's not premature to go ahead and merge.

File: ports/raspberrypi/Makefile (hunk 1 above)
Comment (jepler) [reply ports/raspberrypi/Makefile]: I went ahead and fixed this as I intended. The suggestion in my earlier comment had an error that was probably not easy to spot, so it wouldn't have worked as-is.

**Comment by jepler** [comment]
Ah I see that @bill88t encountered problems when testing with an earlier build of this PR. I did no testing on hardware. So, we should probably have another round of testing before this is considered merge-worthy.

**Comment by bill88t** [comment]
Oh I forgot this existed. I will look into it tomorrow.

**Comment by jepler** [comment]
Thanks, let us know what you find in this round of testing!

**Review by jepler — APPROVED** [review]
Thanks!

**Pull request merged**